src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeWatcher.cs

```diff
@@ -189,10 +234,16 @@ public void Dispose()
 
             public IFileWatchingToken EnqueueWatchingFile(string filePath)
             {
-                // If we already have this file under our path, we don't have to do additional watching
-                if (_directoryFilePath != null && filePath.StartsWith(_directoryFilePath))
+                // If we already have this file under our path, we may not have to do additional watching
+                foreach (var watchedDirectory in _watchedDirectories)
                 {
-                    return _noOpFileWatchingToken;
+                    if (watchedDirectory != null && filePath.StartsWith(watchedDirectory.Path))
+                    {
+                        if (watchedDirectory.ExtensionFilter == null || filePath.EndsWith(watchedDirectory.ExtensionFilter))
```

Comment:
explain the filter.

Reply:
Comment has been added as a part of https://github.com/dotnet/roslyn/compare/30901c8b5fe0e60b4c8571314c7b35911c6fd72e..504c226121a45505c552f79d1e26b719f65a2823.